pkg/eks/api.go

```diff
@@ -266,34 +266,38 @@ func (c *ClusterProvider) CheckAuth() error {
 }
 
 // EnsureAMI ensures that the node AMI is set and is available
-func (c *ClusterProvider) EnsureAMI(version string, ng *api.NodeGroup) error {
+func EnsureAMI(provider api.ClusterProvider, version string, ng *api.NodeGroup) error {
 	if api.IsAMI(ng.AMI) {
-		return ami.Use(c.Provider.EC2(), ng)
+		return ami.Use(provider.EC2(), ng)
 	}
 
 	var resolver ami.Resolver
 	switch ng.AMI {
 	case api.NodeImageResolverAuto:
-		resolver = ami.NewAutoResolver(c.Provider.EC2())
+		resolver = ami.NewAutoResolver(provider.EC2())
 	case api.NodeImageResolverAutoSSM:
-		resolver = ami.NewSSMResolver(c.Provider.SSM())
+		resolver = ami.NewSSMResolver(provider.SSM())
+	case api.NodeImageResolverStatic:
+		resolver = ami.NewStaticResolver()
 	default:
-		resolver = ami.NewDefaultResolver()
+		resolver = ami.NewMultiResolver(
+			ami.NewSSMResolver(provider.SSM()),
+			ami.NewAutoResolver(provider.EC2()),
```

Comment:
should we have the static one as the last delegate?

Reply:
Auto resolution uses the same set of calls used by the static generation process, so if auto resolution fails, there's a good chance the static lookup will also fail.
Also, the plan is to deprecate static resolution and eventually remove it.